rootfs/scheduler/mock.py

```diff
@@ -274,6 +278,104 @@ def upsert_pods(controller, url):
+        del deployment['status']['updatedReplicas']
+    if 'availableReplicas' in deployment['status']:
+        del deployment['status']['availableReplicas']
+    cache.set(url, deployment, None)
+
+    # hash deployment.spec.template with adler32 to get pod hash
+    pod_hash = str(adler32(bytes(json.dumps(deployment['spec']['template'], sort_keys=True), 'UTF-8')))  # noqa
+
+    # fix up url
+    rs_url = url.replace('_deployments_', '_replicasets_')
+
+    # create new RS
+    rs = deployment.copy()
+    rs['kind'] = 'ReplicaSet'
+    # fix up the name
+    rs['metadata']['name'] = rs['metadata']['name'] + '-' + pod_hash
+    rs_url += '_' + pod_hash
+
+    # add the pod-template-hash label
+    rs['metadata']['labels'] = rs['spec']['template']['metadata']['labels'].copy()
+    rs['metadata']['labels']['pod-template-hash'] = pod_hash
+    rs['spec']['template']['metadata']['labels']['pod-template-hash'] = pod_hash
+
+    # deployment only
+    del rs['spec']['strategy']
+
+    # save new ReplicaSet to cache
+    add_cache_item(rs_url, 'replicasets', rs)
+
+    namespaced_url = rs_url[0:(rs_url.find("_replicasets") + 12)]
+    data = cache.get(namespaced_url, [])
+
+    # spin up/down pods for RS
+    upsert_pods(rs, rs_url)
+
+    # Scale down older ReplicaSets
+    for item in data:
+        # skip latest
+        if item == rs_url:
+            continue
+
+        old_rs = cache.get(item)
+
+        # lame way of seeing if the RSs have the same Deployment parent
+        # have to prune of hash as well
+        deployment_url = item.replace('_replicasets_', '_deployments_').replace('_' + old_rs['metadata']['labels']['pod-template-hash'], '')  # noqa
+        if url != deployment_url:
+            continue
+
+        if old_rs['spec']['replicas'] == 0:
+            continue
+
+        old_rs['spec']['replicas'] = 0
+
+        upsert_pods(old_rs, item)
+
+    # Fill out deployment.status for success as pods transition to running state
+    pod_url = namespaced_url.replace('_replicasets', '_pods').replace('apis_extensions_v1beta1', 'api_v1')  # noqa
+    while True:
+        # The below needs to be done to emulate Deployment handling things
+        # always cleanup pods
+        cleanup_pods()
+        # always transition pods
+        pod_state_transitions()
+
+        current = 0
+        for pod in filter_data({'labels': rs['metadata']['labels']}, pod_url):
+            # when this is in the Mock class this can be _pod_ready call
+            if pod['status']['phase'] == 'Running':
+                current += 1
+
+        deployment['status']['updatedReplicas'] = current
+        deployment['status']['availableReplicas'] = current
+        deployment['status']['unavailableReplicas'] = deployment['spec']['replicas'] - current
+        cache.set(url, deployment, None)
+
+        # all ready and matching the intent
+        if current == deployment['spec']['replicas']:
+            break
+
+        time.sleep(0.5)
```

Comment:
Just the sleep before it starts the next go-around of hitting `cleanup_pods` and `pod_state_transitions`. Needed to do it so the system didn't get out of whack. Can add some comments